include/alpaka/core/ConcurrentExecPool.hpp

```diff
@@ -77,118 +73,74 @@ namespace alpaka::core
 #    pragma clang diagnostic push
 #    pragma clang diagnostic ignored "-Wweak-vtables"
 #endif
-        class ITaskPkg
+        struct ITaskPkg
         {
-        public:
             virtual ~ITaskPkg() = default;
 
             //! Runs this task.
-            auto runTask() noexcept -> void
+            void runTask() noexcept
             {
                 try
                 {
                     run();
                 }
                 catch(...)
                 {
-// Workaround: Clang can not support this when natively compiling device code. See ConcurrentExecPool.hpp.
+// Workaround: Clang can not support this when natively compiling device code.
```

Comment:
Just wondering: is this a problem only for CUDA, or also for HIP and SYCL device compilation ?

Reply:
I have no idea. I could even be that the workaround is no longer necessary. It was added in c7b27ed6afed86da26146d6ec3aacc3e350a1762 in May 2016.

Reply:
Please open an issue for the work-around. We should try to remove it.